src/textual/_xterm_parser.py

```diff
@@ -244,10 +258,20 @@ def _sequence_to_key_events(
         """
         keys = ANSI_SEQUENCES_KEYS.get(sequence)
         if isinstance(keys, tuple):
-            # If the sequence mapped to a tuple, then it's values from the
-            # `Keys` enum. Raise key events from what we find in the tuple.
-            for key in keys:
-                yield events.Key(key.value, sequence if len(sequence) == 1 else None)
+            # If we're being asked to ignore the key...
+            if keys == (Keys.Ignore,):
```

Comment:
Do we need this branch? The data structure implies that there could be `Keys.Ignore` mixed with other keys. If that can't happen then a `None` sentinal would be more canonical.

How about we perform the check for ignores keys inside the for loop on L271

Reply:
This approach is following the [pre-established convention](https://github.com/Textualize/textual/blob/3ab7b57d3b7bcc63bd68a259b0bce60b65c0772e/src/textual/_ansi_sequences.py#L176-L177) of marking an ignored sequence as a tuple of a single `Keys.Ignore`; by that convention `Keys.Ignore` will only be used this way. Then following on from that an event is created that contains the sequence information so it can be logged.

Sure, this could be done inside the loop, but given that `Keys.Ignore` should never appear in a context other than when used as `(Keys.Ignore,)`, it didn't seem to make sense to repeat the comparison within the loop: made more sense to be to check once and be done.

Reply:
Using a sequence like this is always going to imply that you can mix ignored keys with actual keys. I fear future readers of the keys file may make that assumption.

If that will never occur, then I suggest we go with a `None` sentinal. Or we roll it in to the loop, so its no longer a special case.

Reply:
Cool, works for me: was just going with what we discussed yesterday (keeping with the inherited convention). I'll switch it up.